include/aws/http/proxy_strategy.h

```diff
@@ -18,33 +18,28 @@ struct aws_http_header;
 struct aws_http_proxy_strategy;
 struct aws_http_proxy_strategy_factory;
 
-/* function defined in crt to get kerberos usertoken*/
-#ifdef __cplusplus
-extern "C" char* get_kerberos_usertoken();
-#endif
-
-/* function defined in crt to get ntlm credential*/
-#ifdef __cplusplus
-extern "C" char *get_ntlm_credential();
-#endif
-
-/* function defined in crt to get ntlm credential*/
-#ifdef __cplusplus
-extern "C" char *get_ntlm_response();
-#endif
-
-#ifdef __cplusplus
-extern "C" void send_ntlm_challenge_header(
-    size_t length,
-    uint8_t *httpHeader,
-    size_t length1,
-    uint8_t *httpHeader1,
-    size_t num_headers);
-#endif
+/*SA-Added Start*/
+
+/*enum defination for callback state*/
+enum proxy_strategy_callback_state {
+    AWS_KERB_TOKEN,
+    AWS_NTLM_CRED,
+    AWS_NTLM_RESP,
+};
+
+/**
+ * User-supplied transform callback function that sends user data to user 
+ *(example NTLM challenge)
+ */
+typedef void (*aws_http_proxy_send_user_data_callback_fn)(size_t data_length, uint8_t *data, void *userdata);
```

Comment:
"user data" in the comment is confusing since it's definitely not related to the userdata parameter.

Reply:
updated documentation